chaindataloader/binary.go

```diff
@@ -0,0 +1,227 @@
+package chaindataloader
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino/headerfs"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+/*
+
+================================================================================
+
+ENCODING FORMAT
+================================================================================
+
+Each file has a header of varint size consisting of,
+
+dataType || startHeight || endHeight || chain
+
+in that order.
+*/
+
+// binReader is an internal struct that holds all data the binReader needs
+// to fetch headers.
+type binReader struct {
+	// reader represents the source to be read.
+	reader io.ReadSeeker
+
+	// startHeight represents the height of the first header in the file.
+	startHeight uint32
+
+	// endHeight represents the height of the last header in the file.
+	endHeight uint32
+
+	// offset represents the distance required to read the first header from
+	// the file.
+	initialOffset uint32
+
+	// chain represents the bitcoin network the headers in the file belong to.
+	chain        wire.BitcoinNet
+	dataTypeSize uint32
+}
+
+// newBinaryChainDataLoader initializes a Binary Reader.
+func newBinaryChainDataLoader(reader io.ReadSeeker,
+	typeData dataType) (*binReader, error) {
+
+	// Read data type.
+	dataTypeBytes := [8]byte{}
```

Comment:
Style nit, can just be: 
```go
var dataTypeBytes [8]byte
```

Reply:
Also this is just scratch space used for decoding, so we can call it `var scratch [8]byte`.